fe/fe-core/src/main/java/com/starrocks/connector/PartitionUtil.java

```diff
@@ -243,6 +243,25 @@ public static List<String> getFilteredPartitionKeys(List<String> partitionNames,
         return filteredPartitionName;
     }
 
+
+    public static List<PartitionKey> getPartitionKeys(Table table) {
+        List<PartitionKey> partitionKeys = Lists.newArrayList();
+        if (table.isUnPartitioned()) {
+            partitionKeys.add(new PartitionKey());
+        } else {
+            try {
+                for (String partitionName : getPartitionNames(table)) {
+                    partitionKeys.add(
+                            createPartitionKey(toPartitionValues(partitionName), getPartitionColumns(table), table));
```

Comment:
The `getPartitionColumns` can be moved out of the for loop ?

Reply:
Done